### PR TITLE
add feature flag full_derive 

### PR DESCRIPTION
**Pull request by echevrier**
To fix issue https://github.com/paritytech/frame-metadata/issues/39
full_derive cannot be enabled for metadata prior to version v14

### CODE REVIEW

File: frame-metadata/src/lib.rs [hunk 1]
```diff
@@ -20,14 +20,32 @@
...
+		feature = "v12",
+		feature = "v11",
+		feature = "v10",
+		feature = "v9",
+		feature = "v8",
+		feature = "legacy"
+	)
+))]
+compile_error!("feature \"full_derive\" cannot be enabled for metadata prior to version v14");
+
```
Comment (clangenb) [inline frame-metadata/src/lib.rs:36]: I have double-checked, I thought that the old metadata versions are not being completely supported in terms of decoding and serializing. However, it seems that they are. If we really want to do an upstream PR, I believe we can't neglect the old ones. So I need to ask you to repeat the pattern from metadata v14 for the old versions.

File: frame-metadata/Cargo.toml [hunk 2]
```diff
@@ -15,11 +15,11 @@ targets = ["x86_64-unknown-linux-gnu"]
...
 codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive"] }
 cfg-if = "1.0.0"
-scale-info = { version = "2.0.0", default-features = false, optional = true, features = ["derive"] }
-serde = { version = "1.0.101", optional = true, features = ["derive"] }
+scale-info = { version = "2.0.0", default-features = false, optional = true, features = ["derive", "decode"] }
+serde = { version = "1.0.101", default-features = false, features = ["alloc", "derive"] }
 
 [features]
-default = ["std", "v14"]
+default = [ "std", "v14"]
```
Comment (clangenb) [inline frame-metadata/Cargo.toml:22]: minor fmt

**Review by clangenb — COMMENTED** [review]
The pattern looks perfect, but if we want to supply an upstram PR, I thnink we need to do some more work. :)

File: frame-metadata/src/lib.rs (hunk 1 above)
Comment (echevrier) [reply frame-metadata/src/lib.rs:36]: Metadata prior V14 do not support `Decode` if no_std.  See https://github.com/integritee-network/frame-metadata/blob/d241ab9c1ba89e119263812f963a601be158e661/frame-metadata/src/decode_different.rs#L29 
For now the std flag set "full" derives in metadata prior V14. Do we really want to differentiate between the derive levels or between the technical possible ones?  
We added the full-derive flag for v14 to get the derives decode, serialize,... in no_std as they are technically feasible with scale-info. For the metadata prior V14 we could have the Serialize in no_std, but Decode only in std. Does it make sense to have this differentiation?

Comment (clangenb) [reply frame-metadata/src/lib.rs:36]: Ahbok, if it is like that, I agree with you. Let's leave it like that. Sorry for that!

Comment (echevrier) [reply frame-metadata/src/lib.rs:36]: The changes in this PR are wrong. So if we only change V14, please check https://github.com/integritee-network/frame-metadata/pull/2

**Comment by clangenb** [comment]
Closed in favor of #2

**Pull request closed without merging**